### PR TITLE
Add documentation and validations

### PR DESCRIPTION
**Pull request by erwinc1**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,13 +23,14 @@ For use with Spring Boot applications. See [Shadow Tool](https://github.com/rabo
...
+|------------------------------------------------------|--------------------------------------------------------------|---------------|-------------------------------------------------------------------------------------------------------------------------------------------------|
+| `shadowflow.encryption.cipher.secret`                | `String`                                                     | `""`          | The secret for encryption. Should be a 16, 24, or 32-byte string. Could be generated as follows: `openssl rand -hex 32`                         |
+| `shadowflow.encryption.cipher.initialization-vector` | `String`                                                     | `""`          | The initialization vector for encryption. Should be a 12-byte string. Could be generated as follows: `openssl rand -hex 12`                     |
+| `shadowflow.encryption.public-key`                   | `String`                                                     | `""`          | Base 64 encoded version of an `X509` Public Key. Used in a Cipher with algorithm `RSA/ECB/OAEPWITHSHA-256ANDMGF1PADDING`.                       |
```
Comment (martinvisser) [inline README.md:30]: Technically speaking the default is just `null`, not an empty string.
```suggestion
| `shadowflow.encryption.cipher.secret`                | `String`                                                     | `null`          | The secret for encryption. Should be a 16, 24, or 32-byte string. Could be generated as follows: `openssl rand -hex 32`                         |
| `shadowflow.encryption.cipher.initialization-vector` | `String`                                                     | `null`          | The initialization vector for encryption. Should be a 12-byte string. Could be generated as follows: `openssl rand -hex 12`                     |
| `shadowflow.encryption.public-key`                   | `String`                                                     | `null`          | Base 64 encoded version of an `X509` Public Key. Used in a Cipher with algorithm `RSA/ECB/OAEPWITHSHA-256ANDMGF1PADDING`.                       |
```

File: README.md [hunk 2]
```diff
@@ -37,17 +39,20 @@ file. Here are the steps to configure shadow flows:
 
 1. Define the shadow flows you want to use in your application under the `shadowflow.flows` property.
    In this example, `flow1` and `flow2` are defined as shadow flows (which you can rename to anything).
-   And `percentage` represents a percentage number in the range of 0-100 of how many calls should be compared in the
+   `percentage` represents a percentage number in the range of 0-100 of how many calls should be compared in the
```
Comment (martinvisser) [inline README.md:42]:
```suggestion
   `percentage` represents a number in the range of 0-100 of how many calls should be compared in the
```

File: README.md [hunk 3]
```diff
@@ -37,17 +39,20 @@ file. Here are the steps to configure shadow flows:
...
    shadow flow.
    Zero effectively disables the shadow flow.
+   And `type` represents the data model which is used to compare.
```
Comment (martinvisser) [inline README.md:45]:
```suggestion
   shadow flow. \
   Zero effectively disables the shadow flow. \
   And `type` represents the data model which is used to compare.
```

Or did you mean to put it display it on one line?

Comment (erwinc1) [reply README.md:45]: Yeah, I meant it to be displayed on one line. I added the return for readability in IDEs

**Review by martinvisser — APPROVED** [review]

**Pull request merged**